packages/flutter_tools/templates/list_detail_app/lib/src/app.dart.tmpl

```diff
@@ -0,0 +1,69 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:{{projectName}}/src/dummy_feature/dummy_item_details_view.dart';
+import 'package:{{projectName}}/src/dummy_feature/dummy_item_list_view.dart';
+import 'package:{{projectName}}/src/settings/settings_controller.dart';
+import 'package:{{projectName}}/src/settings/settings_view.dart';
+
+/// The Widget that configures your application.
+class MyApp extends StatelessWidget {
+  const MyApp({
+    Key key,
+    @required this.settingsController,
+  }) : super(key: key);
+
+  final SettingsController settingsController;
+
+  @override
+  Widget build(BuildContext context) {
+    // Glue the SettingsController to the MaterialApp.
+    //
+    // The AnimatedBuilder Widget listens to the SettingsController for changes.
+    // Whenever the user updates their settings, the MaterialApp is rebuilt.
+    return AnimatedBuilder(
+      animation: settingsController,
+      builder: (context, _) {
```

Comment:
Please give full types and names for every argument to every closure, that way it's self-documenting. Otherwise, a developer looking at this will have no idea what the second argument is nor what the type of the first one is.